test/transactionRejection.js

```diff
@@ -153,27 +135,17 @@ describe("Transaction rejection", function() {
           return done(new Error("Error was returned which had no message"));
         }
       } else if (response.result) {
-        web3.eth
-          .getTransactionReceipt(response.result)
-          .then((result) => {
-            if (to.number(result.status) === 0) {
-              return done(
-                new Error("TX rejections should return error, but returned receipt with zero status instead")
-              );
-            } else {
-              return done(
-                new Error(
-                  `TX should have rejected prior to running. Instead transaction ran successfully (receipt.status == 
-                    ${to.number(result.status)})`
-                )
-              );
-            }
-          })
-          .catch((error) => {
-            return done(error);
-          });
+        const result = await web3.eth.getTransactionReceipt(response.result);
+        if (to.number(result.status) === 0) {
+          assert.fail("TX rejections should return error, but returned receipt with zero status instead");
```

Comment:
Since this test hasn't been `async`/`await` ed, but still executes asynchronously it needs to call `done` even in the case of an error. The test will still fail with these `assert.fail` calls, but they'll just take the entire timeout length to do so. And the error will be from the timeout, and we'll _also_ see an `UnhandledPromiseRejectionWarning` due to the `assert.fail` throwing an exception.

All that to say: either promisify this and fully convert to async/await (no mocha `done` should be used) or put the `done(new Error(...` statements back.

This also no longer handles the exceptional case where `getTransactionReceipt` throws (`.catch((error)`), like the test used to on lines 172-174.